public/app/features/variables/inspect/utils.ts

```diff
@@ -419,3 +367,24 @@ export function flattenPanels(panels: PanelModel[]): PanelModel[] {
 
   return result;
 }
+
+// Accepts an array of panel models, and returns an array of panel IDs paired with
+// the names of any template variables found
+export function getPanelVars(panels: PanelModel[]) {
+  const panelsByVar: Record<string, Set<number>> = {};
+  for (const p of panels) {
+    const jsonString = stringifyPanelModel(p);
+    const repeats = [...jsonString.matchAll(/"repeat":"([^"]+)"/g)].map((m) => m[1]);
```

Comment:
I would feel a bit more comfortable if we explicitly used the PanelModel to get the repeat variables. That way we get actual type safety, and can accurately answer questions like "hey, where do we use this `repeat` property?"